Chunks/ChunkManagement/ChunkWorker.cs

```diff
@@ -41,30 +43,53 @@ private enum JobType
 
     private readonly ChunkFactory _chunkFactory;
 
-    private const int RenderDistance = 2;
+    private const int RenderDistance = 1;
+
+    private const int NumberOfThreads = 2;
 
-    private const int NumberOfThreads = 1;
+    private static readonly Logger Logger = LogManager.GetCurrentClassLogger();
+
+    private readonly ChunkHandler _chunkHandler;
 
     private static int CacheNumber => 2 * (2 * RenderDistance + 1) * (2 * RenderDistance + 1) * (2 * RenderDistance + 1);
 
-    private readonly CancellationTokenSource _cancellationTokenSource = new();
+    private CancellationTokenSource _cancellationTokenSource = new();
 
-    public ChunkWorker(List<Chunk> chunks, SimulationManager<PoseIntegratorCallbacks> simulationManager, ChunkFactory chunkFactory)
+    public ChunkWorker(List<Chunk> chunks, SimulationManager<PoseIntegratorCallbacks> simulationManager, Settings settings)
     {
         _chunks = chunks;
         _simulationManager = simulationManager;
-        _chunkFactory = chunkFactory;
+        _chunkFactory = new ChunkFactory(new ScalarFieldGenerator(settings.Seed));
+        _chunkHandler = new ChunkHandler(settings.SaveName);
         foreach (var chunk in _chunks)
         {
             _existingChunks.Add(chunk.Position / Chunk.Size);
         }
 
-        Directory.CreateDirectory(ChunkHandler.SaveLocation);
+        Start();
+    }
 
+    private void Start()
+    {
+        _cancellationTokenSource = new CancellationTokenSource();
         for (int i = 0; i < NumberOfThreads; i++)
         {
             Task.Run(RunJobs);
         }
+
+        GetSavedChunks();
+        EnqueueLoadingChunks(Vector3i.Zero);
+        int totalChunks = (2 * RenderDistance + 1) * (2 * RenderDistance + 1) * (2 * RenderDistance + 1);
+        int prevNumber = 0;
+        while (_chunks.Count < totalChunks)
```

Comment:
I'm not sure I get it. Could we discuss it later?

Reply:
sure